### PR TITLE
refactoring and fixes

### PR DESCRIPTION
**Pull request by roipoussiere**

### CODE REVIEW

**Comment by roipoussiere** [comment]
I think it's enough for this PR.

This one is not atomic at all because there was various things to modify, but commits are supposed to be atomic. That's why I suggest to review this PR commit by commit instead file by file.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,2 @@
-# Dir for bundles
-dist
```
Comment (guhur) [inline .gitignore:2]: Why did you remove it? This is a build folder right?

File: mv-openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,476 @@
+# Warning: this file has been automatically generated. All manual modifications will be lost, see README.md for details.
```
Comment (guhur) [inline mv-openapi.yaml:1]: If this file is automatically generated it should be removed from the commits?

File: mv-openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+      type: http
+      scheme: bearer
+    pollAuth:
+      type: http
+      scheme: bearer
+    resultAuth:
+      type: http
+      scheme: bearer
+  schemas:
+    ballot:
```
Comment (guhur) [inline mv-openapi.yaml:268]: I read somewhere that schemas should start with an uppercase in openapi

File: mv-openapi.yaml [hunk 4]
```diff
@@ -0,0 +1,476 @@
...
+              schema:
+                $ref: '#/components/schemas/result'
+        '404':
+          $ref: '#/components/responses/poll-not-found'
+components:
+  securitySchemes:
+    participantAuth:
+      type: http
+      scheme: bearer
+    pollAuth:
```
Comment (guhur) [inline mv-openapi.yaml:261]: Do you want to use kebab case?

File: mv-openapi.yaml [hunk 5]
```diff
@@ -0,0 +1,476 @@
...
+        name: ballot-id
+        required: true
+        schema:
+          type: string
+    get:
+      operationId: get-ballot
+      summary: Gets a ballot by ID
+      responses:
+        '200':
+          description: A ballot object
```
Comment (guhur) [inline mv-openapi.yaml:166]: I think it's way more standard to use the usual http code OK. I've looked at several API. I don't think any tooling script would fail though with a non standard message.

File: .gitignore (hunk 1 above)
Comment (roipoussiere) [reply .gitignore:2]: I updated the package config to generate the openapi file in the root folder (just like the old, hand-written one). So the dist folder is not necessary.

File: mv-openapi.yaml (hunk 2 above)
Comment (roipoussiere) [reply mv-openapi.yaml:1]: I don't think so.

It can be really usefull to have to main openapi file located somewhere in this repository. For instance if you want put this file on a web server you just have to wget it, not to clone the repo, install npm and run the build.

We just have to don't forget to build it before to push, but this can be tested with a CI tool or git hooks.

File: mv-openapi.yaml (hunk 5 above)
Comment (roipoussiere) [reply mv-openapi.yaml:166]: On which line this comment belongs to exactly?

File: mv-openapi.yaml (hunk 4 above)
Comment (roipoussiere) [reply mv-openapi.yaml:261]: On which line this comment belongs to exactly?

File: mv-openapi.yaml (hunk 3 above)
Comment (roipoussiere) [reply mv-openapi.yaml:268]: Ok I will investigate.

Comment (guhur) [reply mv-openapi.yaml:268]: Hum weird I can't find it. Then it is certainly not a super important convention!

File: mv-openapi.yaml (hunk 5 above)
Comment (guhur) [reply mv-openapi.yaml:166]: Good point. It seems that no one (and thus tooling) about it. So let's leave it as you did.

File: mv-openapi.yaml (hunk 4 above)
Comment (guhur) [reply mv-openapi.yaml:261]: let's do it on another PR
